include/SFML/Window/Event.inl

```diff
@@ -0,0 +1,64 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2023 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+Event::Event(const T& t)
+{
+    static_assert(isEventType<T>, "T must be a subtype of sf::Event");
+    if constexpr (isEventType<T>)
```

Comment:
Probably missing something, but why have the `if constexpr` at all if the `static_assert` above performs the same check?

Reply:
I addressed that previously in this thread. TL;DR It allows for better compiler diagnostics when users do something incorrect.

Reply:
> I addressed that previously in this thread. TL;DR It allows for better compiler diagnostics when users do something incorrect.

Probably worth adding a comment in the code itself, I can imagine a future maintainer having the same question and possibly removing the redundant-appearing checks.

Reply:
I don't plan on abandoning the project plus I read every PR so I'm not particularly worried about someone removing this without me realizing it. A comment isn't a bad idea though.

Reply:
@ChrisThrasher: ditto, you seemed in favour of the idea of adding a comment, and the conversation was resolved, yet I do not see the suggestion applied.

Reply:
This is still open; adding a comment would help this seeming duplication.

Reply:
There’s already a comment about this higher up in the file